vim-patch:8.2.4133: output of ":scriptnames" goes into the message history

zeertzjq opened the pull request:
#### vim-patch:8.2.4133: output of ":scriptnames" goes into the message history

Problem:    output of ":scriptnames" goes into the message history, while this
            des not happen for other commands, such as ":ls".
Solution:   Use msg_outtrans() instead of smsg().
https://github.com/vim/vim/commit/840f16202e1ae2d574507ef52a7e8a98775f243c

bfredl commented on src/nvim/ex_cmds2.c:
```diff
@@ -2323,9 +2324,12 @@ void ex_scriptnames(exarg_T *eap)
+      msg_putchar('\n');
+      msg_outtrans(IObuff);
+      ui_flush();  // output one line at a time
+      os_breakcheck();
```
`line_breakcheck()`

bfredl commented on src/nvim/ex_cmds2.c:
```diff
@@ -2323,9 +2324,12 @@ void ex_scriptnames(exarg_T *eap)
+      vim_snprintf((char *)IObuff, IOSIZE, "%3d: %s", i, NameBuff);
+      msg_putchar('\n');
+      msg_outtrans(IObuff);
+      ui_flush();  // output one line at a time
```
does it work properly without `ui_flush()` ? ui_flush should already be emitted by message.c precisely when needed, while avoid unnecessary flicker.

zeertzjq replied:
Removing `ui_flush()` indeed seems to still work properly and is faster.

zeertzjq replied:
However, it seems `ui_flush()` is used in many places that output to messages.

bfredl replied:
yes, and in many of those places it can likely be removed :)

zeertzjq committed:
vim-patch:8.2.4133: output of ":scriptnames" goes into the message history

Problem:    output of ":scriptnames" goes into the message history, while this
            des not happen for other commands, such as ":ls".
Solution:   Use msg_outtrans() instead of smsg(). (closes vim/vim#9551)
https://github.com/vim/vim/commit/840f16202e1ae2d574507ef52a7e8a98775f243c
